### PR TITLE
feat(pkce): add ability for PKCE clients to use refresh_tokens

### LINKED ISSUE #472: Allow token refresh using the PKCE flow
Need to handle the case here: https://github.com/mozilla/fxa-oauth-server/blob/52d935183b7bbd5813f52def13ac14c22acd162a/lib/routes/token.js#L177 where clients don't have a client_secret and want to renew their poor token 💻

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #472

@rfk @seanmonstar r?

### CODE REVIEW

File: lib/routes/token.js [hunk 1]
```diff
@@ -175,9 +179,31 @@ module.exports = {
...
+          var confirmClientPromise;
+
+          if (dbClient.publicClient) {
+            if (params.client_secret) {
+              throw new AppError.invalidRequestParameter('client_secret');
+            }
+
+            confirmClientPromise = confirmPkceClient(params.client_id);
+          } else {
+            confirmClientPromise = confirmClient(params.client_id, params.client_secret);
```
Comment (seanmonstar) [inline lib/routes/token.js]: The `confirmClient` does another db lookup, but this function already did one, and then compares a hash of the secrets.

**Comment by vladikoff** [comment]
> It seems to me that the confirmBlahClient stuff could be refactored ever so slightly. So as to not have secret comparison code duplicated, that could just be 1 function, used both here and in the original confirmClient function.

Thanks, that's a good plan!

**Comment by vladikoff** [comment]
@seanmonstar How about now? I tried moved the db code out of the existing functions. Let me know..

**Comment by vladikoff** [comment]
Example usage: 

POST http://127.0.0.1:9010/v1/token

```
{
  "refresh_token": "8425d9248b92f86d7bf518ee6dc51e304c8702892bbe2466e487459c05d64628",
  "client_id": "c6d74070a481bc10",
  "grant_type": "refresh_token",
  "scope": "profile%20https%3A%2F%2Fidentity.mozilla.org%2Fapps%2Fnotes"
}
```

**Pull request merged**